Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,82 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
-      end
-    end
+    # TODO: goal is to no longer require def commands at all
+    CommandsCommand.call
   end
 
-  def internal_commands
-    find_internal_commands HOMEBREW_LIBRARY_PATH/"cmd"
-  end
+  class CommandsCommand < Command
+    options do
+      command "commands"
+      desc "Install <formula>."
+      argument :formulae
+      option "d", "debug", desc: "open an interactive debugging session with access to IRB or a shell inside the temporary build directory."
```

Comment:
I think it would be good to just have a `flag: "d"` or similar for the things with short options rather than taking two string parameters.